### ISSUE TITLE
Enable qr code import

### ISSUE THREAD

**Issue by alexhauser** [sqrldev/SQRLDotNetClient#208]
### Description:
Enable importing a SQRL identity by scanning its QR code using the system's default web cam / capture device. 

I'm currently working on this one, @josegomez.

**Comment by josegomez** [comment]
Cool!

**Comment by alexhauser** [comment]
@josegomez, short status report:

The whole thing started working flawlessly on Windows in no time:
<img src="https://user-images.githubusercontent.com/4005543/84515462-ca909400-accc-11ea-87f5-82e0098ccc29.gif" width="300">

Then, guess what?! I began testing on Linux and macOS and everything started falling apart. 😆

**Comment by josegomez** [comment]
yeah... linux our old "friend" @alexhauser  LoL
Let me know if you need a hand. I was looking for something todo but frankly the open issues other than the one you are working on are either not our problem or  non fixable.

**Comment by alexhauser** [comment]
I don't know... I'm currently using [OpenCVSharp](https://github.com/shimat/opencvsharp) for capturing frames from the webcam, but I have problems getting it running on Linux and haven't even tried macOS, since we have a general libgdiplus problem there (even with `mono-libgdiplus` installed our qr code import/export code fails at the moment - see https://github.com/dotnet/runtime/issues/27200).

I am leaning towards creating our own platform specific implementations for captruring frames from a video device (using our proven Interface / Implementation model) and coding everything ourselves instead of relying on heavyweight and fragile external libraries.

If that is something that sparks your interest (e.g. writing the Linux/macOS implementation for that), be my guest! 😁

**Comment by josegomez** [comment]
hhaha nothing sounds more exciting than writing platform specific stuff. I dream of fighting with Linux and Mac in Particular
All joking aside I don't mind so if this is the way you think we should go throw me an interface and I'll take a crack at it.

**Comment by alexhauser** [comment]
Cool, Jose! 👏

However, you may want to take a look at the current state of affairs first and evaluate yourself, since I might be overreacting ;-) 

You could checkout the `import_qr` branch on my fork at https://github.com/alexhauser/SQRLDotNetClient and poke around a little if you like.

**Comment by josegomez** [comment]
@alexhauser i'm still here sorry I've been awol, life has gotten in the way but this is still in my #TODO!

**Comment by alexhauser** [comment]
No worries @josegomez, we're absolutely not in a hurry!

**Comment by dxcSithLord** [comment]
PyImageSearch has several blogs on QR detection code and images - have a look at this post, using Python, but the algorithm should be the same in .net:
[https://www.pyimagesearch.com/2018/05/21/an-opencv-barcode-and-qr-code-scanner-with-zbar/](https://www.pyimagesearch.com/2018/05/21/an-opencv-barcode-and-qr-code-scanner-with-zbar/)

**Comment by alexhauser** [comment]
@dxcSithLord, thanks for the input. However, "the algorightm" is not the issue here. I know how to use the OpenCV library, and detecting barcodes is not a problem, either. The problem is, that the .NET port of the OpenCV library that we rely on for .NET, while working properly on Windows, currently fails to work on Linux and macOS.

**Comment by josegomez** [comment]
@alexhauser  are we just waiting to finish this to release the next version I suppose? I'll try to work some on it this week. The ole J O B has been relentless lately.

**Comment by alexhauser** [comment]
Yes @josegomez, but we can easily release without this one if we want. It's only an enhancement after all, nothing important! Your call! 😊

**Comment by josegomez** [comment]
@alexhauser  we can wait is not like anyone is banging on our doors to finish this... I'm pretty sure unless a miracle happens SQRL in general is DOA :'(

**Comment by alexhauser** [comment]
Amen, @josegomez.
